modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/AbstractIncidentService.java

```diff
@@ -279,15 +277,13 @@ public Incident apply(IncidentDto dto) {
    * "org.opencastproject.composer.1.title"]
    */
   public static List<String> genDbKeys(List<String> locale, String base) {
-    final List<String> keys = mlist(locale).foldl(Mutables.list(base),
-            new Function2<List<String>, String, List<String>>() {
-              @Override
-              public List<String> apply(List<String> sum, String s) {
-                sum.add(sum.get(sum.size() - 1) + "." + s);
-                return sum;
-              }
-            });
-    return mlist(keys).reverse().value();
+    List<String> result = new ArrayList<>();
```

Comment:
If you want to micro-optimize here: `LinkedList` plus `StringBuilder`. But unless this functions is called very often, it shouldn't matter.